File: week1/day5-best-time-to-buy-and-sell-stock-ii.java
/*
    https://leetcode.com/problems/best-time-to-buy-and-sell-stock-ii/
*/

class Solution {
    public int maxProfit(int[] prices) {
        int accum_sum = 0;
        for (int i = 1; i < prices.length; i++) {
            if (prices[i] > prices[i - 1]) {
                accum_sum += prices[i] - prices[i - 1];
            }
        }
        return accum_sum;
    }
}

File: week1/day5-best-time-to-buy-and-sell-stock-ii.py
'''
    https://leetcode.com/problems/best-time-to-buy-and-sell-stock-ii
'''


class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        accum_sum = 0
        for i in range(1, len(prices)):
            prev_px = prices[i-1]
            curr_px = prices[i]
            accum_sum += max(curr_px - prev_px, 0)
        return accum_sum
